dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-dingtalk-personalwork/src/main/java/org/apache/dolphinscheduler/plugin/alert/dingtalk/personalwork/DingTalkPersonalWorkSender.java

```diff
@@ -0,0 +1,340 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.plugin.alert.dingtalk.personalwork;
+
+import org.apache.dolphinscheduler.alert.api.AlertResult;
+import org.apache.dolphinscheduler.common.utils.JSONUtils;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * <p>
+ *     https://open.dingtalk.com/document/orgapp/enterprise-created-chatbot
+ *     https://open.dingtalk.com/document/orgapp/chatbot-sends-queries-and-withdraws-one-on-one-chat-messages
+ * </p>
+ */
+@Slf4j
+public class DingTalkPersonalWorkSender {
```

Comment:
Same as https://github.com/apache/dolphinscheduler/pull/14982#discussion_r1361751343

Reply:
> Same as [#14982 (comment)](https://github.com/apache/dolphinscheduler/pull/14982#discussion_r1361751343)

If we do this, their parameters will appear confusing when placed in a form. Is there a better way?  I think the best user experience is that after modifying `sendtype`, the form parameters will also be modified.

![image](https://github.com/apache/dolphinscheduler/assets/42763775/4830849d-892e-4c9d-935b-954d434e40a8)
![image](https://github.com/apache/dolphinscheduler/assets/42763775/0e508f46-a23d-4c94-bcd0-11a5ed77745e)

Reply:
> > Same as [#14982 (comment)](https://github.com/apache/dolphinscheduler/pull/14982#discussion_r1361751343)
> 
> If we do this, their parameters will appear confusing when placed in a form. Is there a better way? I think the best user experience is that after modifying `sendtype`, the form parameters will also be modified.
> 
> ![image](https://user-images.githubusercontent.com/42763775/278339361-4830849d-892e-4c9d-935b-954d434e40a8.png) ![image](https://user-images.githubusercontent.com/42763775/278339476-0e508f46-a23d-4c94-bcd0-11a5ed77745e.png)

@SbloodyS please check this and #https://github.com/apache/dolphinscheduler/pull/14982#issuecomment-1777144300